File: BOJ_1011/BOJ_1011.cs
using System;
using System.Text;
namespace asharien{
    class PROGRAM{
        static void Main(){
            StringBuilder SB = new StringBuilder();
            int N = int.Parse(Console.ReadLine());
            for(int i=0; i<N; i++){
                long[] DATA = Console.ReadLine().Split().Select(long.Parse).ToArray();
                SB.AppendLine(Convert.ToString(Math.Truncate(Math.Sqrt((4*(DATA[1]-DATA[0]))-1))));
            }
            Console.Write(SB);
        }
    }
}

File: BOJ_1011/BOJ_1011.java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;
import java.math.*;
class Main {
	public static void main(String[] args) 
	throws IOException{
		BufferedReader BR = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter BW = new BufferedWriter(new OutputStreamWriter(System.out));
		StringBuilder SB =  new StringBuilder();
		int N = Integer.parseInt(BR.readLine());
		for(int i=0; i<N; i++){
			String[] X = BR.readLine().split(" ");
			long A = Long.parseLong(X[0]);
			long B = Long.parseLong(X[1]);
			SB.append(String.format("%.0f", Math.floor((Math.sqrt(4*(B-A)-1))))).append("\n");
		}
		BW.write(SB+"");
		BW.flush();
	}
}
